src/client/java/minicraft/core/Renderer.java

```diff
@@ -102,7 +103,7 @@ public static void initScreen() {
 
 		image = new BufferedImage(WIDTH, HEIGHT, BufferedImage.TYPE_INT_RGB);
 		screen.pixels = ((DataBufferInt) image.getRaster().getDataBuffer()).getData();
-		hudSheet = new LinkedSprite(SpriteType.Gui, "hud");
+		hudSheet = () -> Renderer.spriteLinker.getSheet(SpriteType.Gui, "hud");
```

Comment:
Please avoid complicated expressions like this. It would be better to just store the sheet in memory.

Reply:
The sheet object is changed in the linker when changes applied in resources.

Reply:
Could you profile the performance impact of before and after this pull request?

Reply:
This change (on this line) relates to the mechanics of `LinkedSprite`. A `LinkedSprite` updates on demand (when changes of resources applied, valid `LinkedSprite`s update), but it is expected that a `Sprite` is used instead of a sheet.

Reply:
In thread time, it changes from around 920±50 ms/s to around 300±150 ms/s on main thread with the same (initialized) world.
In memory (allocated bytes) per thread, it changes from 9±0.4 MB to around 9.8±0.4 MB on main thread.
These are measured with simple movements and at rest on the same (initialized) world.

Reply:
How often is this lambda function called?

Reply:
Actually, this lambda acts as a wrapper. Without this wrapper, `Renderer.spriteLinker.getSheet(SpriteType.Gui, "hud")` would still be required on each line rendering occuring.

Reply:
That doesn't answer my question.

Reply:
Every time sprites within `hud.png` being rendered.